rsocket-transport-netty/src/main/java/io/rsocket/transport/netty/server/TcpServerTransport.java

```diff
@@ -94,26 +94,30 @@ public static TcpServerTransport create(TcpServer server) {
   @Override
   public Mono<CloseableChannel> start(ConnectionAcceptor acceptor, int mtu) {
     Objects.requireNonNull(acceptor, "acceptor must not be null");
-
-    return server
-        .doOnConnection(
-            c -> {
-              c.addHandlerLast(new RSocketLengthCodec());
-              DuplexConnection connection;
-              if (mtu > 0) {
-                connection =
-                    new FragmentationDuplexConnection(
-                        new TcpDuplexConnection(c, false),
-                        ByteBufAllocator.DEFAULT,
-                        mtu,
-                        true,
-                        "server");
-              } else {
-                connection = new TcpDuplexConnection(c);
-              }
-              acceptor.apply(connection).then(Mono.<Void>never()).subscribe(c.disposeSubscriber());
-            })
-        .bind()
-        .map(CloseableChannel::new);
+    return FragmentationDuplexConnection.checkMtu(mtu)
```

Comment:
This check could be done on RSocketFactory, added here for parity with client signalling error on connect.